pkg/controller/endpointslice/reconciler_test.go

```diff
@@ -119,17 +116,20 @@ func TestReconcile1Pod(t *testing.T) {
 		expectedEndpoint         discovery.Endpoint
 		expectedLabels           map[string]string
 		expectedEndpointPerSlice map[discovery.AddressType][]discovery.Endpoint
-		terminatingGateEnabled   bool
 	}{
 		"no-family-service": {
 			service: noFamilyService,
 			expectedEndpointPerSlice: map[discovery.AddressType][]discovery.Endpoint{
 				discovery.AddressTypeIPv4: {
 					{
-						Addresses:  []string{"1.2.3.4"},
-						Conditions: discovery.EndpointConditions{Ready: utilpointer.BoolPtr(true)},
-						Zone:       utilpointer.StringPtr("us-central1-a"),
-						NodeName:   utilpointer.StringPtr("node-1"),
+						Addresses: []string{"1.2.3.4"},
+						Conditions: discovery.EndpointConditions{
+							Ready:       utilpointer.BoolPtr(true),
+							Serving:     utilpointer.BoolPtr(true),
```

Comment:
If these were not set before, does this change the nature of the tests (defaults to false) ?  Do we need to modulate these fields for full test coverage?

Reply:
Good point -- let me check the existing test coverage and make sure we're adding cases for terminating endpoints in places we previously assumed it was disabled

Reply:
Reviewed tests coverage again -- I think by addiing the serving/terminating condition checks in all the tests that previously had the gate disabled and keeping the tests for terminating endpoints when the gate was enabled keeps our test coverage more or less the same as before. 

I pushed a change to clean up some left over references to the feature gate in the test names though.